src/controls/experimental/CameraControls.ts

```diff
@@ -0,0 +1,1176 @@
+  // Range is 0 to Math.PI radians.
+  minPolarAngle: number // radians
+  maxPolarAngle: number // radians
+
+  // How far you can orbit horizontally, upper and lower limits.
+  // If set, must be a sub-interval of the interval [ - Math.PI, Math.PI ].
+  minAzimuthAngle: number // radians
+  maxAzimuthAngle: number // radians
+
+  // Set to true to enable damping (inertia)
+  // If damping is enabled, you must call controls.update() in your animation loop
+  enableDamping: boolean
+  dampingFactor: number
+
+  // This option enables dollying in and out; property named as "zoom" for backwards compatibility
+  // Set to false to disable zooming
+  enableZoom: boolean
+  zoomSpeed: number
+
+  // Set to false to disable rotating
+  enableRotate: boolean
+  rotateSpeed: number
+
+  // Set to false to disable panning
+  enablePan: boolean
+  panSpeed: number
+  screenSpacePanning: boolean // if true, pan in screen-space
+  keyPanSpeed: number // pixels moved per arrow key push
+
+  // Set to true to automatically rotate around the target
+  // If auto-rotate is enabled, you must call controls.update() in your animation loop
+  // auto-rotate is not supported for trackball behavior
+  autoRotate: boolean
+  autoRotateSpeed: number // 30 seconds per round when fps is 60
+
+  // Set to false to disable use of the keys
+  enableKeys: boolean
+
+  // The four arrow keys
+  keys: { LEFT: number; UP: number; RIGHT: number; BOTTOM: number }
+
+  // Mouse buttons
+  mouseButtons: {
+    LEFT: MOUSE
+    MIDDLE: MOUSE
+    RIGHT: MOUSE
+  }
+
+  // Touch fingers
+  touches = { ONE: TOUCH.ROTATE, TWO: TOUCH.DOLLY_PAN }
+
+  // // for reset
+  target0: Vector3
+  position0: Vector3
+  quaternion0: Quaternion
+  zoom0: number
+
+  // current position in spherical coordinates
+  spherical: Spherical
+  sphericalDelta: Spherical
+
+  private changeEvent: { type: 'change' }
+  private startEvent: { type: 'start' }
+  private endEvent: { type: 'end' }
+  private state: STATE
+
+  private EPS: number
+
+  private scale: number
+  private panOffset: Vector3
+  private zoomChanged: boolean
+
+  private rotateStart: Vector2
+  private rotateEnd: Vector2
+  private rotateDelta: Vector2
+
+  private panStart: Vector2
+  private panEnd: Vector2
+  private panDelta: Vector2
+
+  private dollyStart: Vector2
+  private dollyEnd: Vector2
+  private dollyDelta: Vector2
+
+  constructor(object: PerspectiveCamera | OrthographicCamera, domElement: HTMLElement) {
+    super()
+
+    if (domElement === undefined) {
+      console.warn('THREE.CameraControls: The second parameter "domElement" is now mandatory.')
+    }
+    if (domElement instanceof Document) {
+      console.error(
+        'THREE.CameraControls: "document" should not be used as the target "domElement". Please use "renderer.domElement" instead.',
+      )
+    }
+
+    this.object = object
+    this.domElement = domElement
+
+    // Set to false to disable this control
+    this.enabled = true
```

Comment:
things like these can be moved out of the constructor, it creates noise imo

Reply:
so initialize them above the constructor like this? https://github.com/pmndrs/three-stdlib/commit/64768bcbb5698cc526c7cbe52b4f42b3d4d4f1f4#diff-7033244187fc57a9f7b3894307ebef2fa9b101cf3541983542df182ee5a5cc0cR7

Reply:
aaaaaand got it!